Add dataform runtime validation (#864)
telerik/maui-docs: controls/dataform/validation.md

![.NET MAUI DataForm Error styling](images/dataform-error-message-styling-desktop.png)

## Runtime Validation

If you want to add the validation runtime, when editors are generated, then you have to use the `EditorsGenerated` event. Inside the event you can define the validation. 

> The available validations DataForm provides are listed in the table:

Reviewer: ```suggestion
The following table lists the available DataForm validations:
```

| DataForm Validation Rule | Definition |
| ----- | ------ |
| (DataForm level) `DataFormObjectValidationRule` | Represents a custom validation rule of the DataForm control. |
| `DataFormEditorCustomValidationRule` | Represents a custom validation rule for the editor. The validation rule can execute a custom validation logic on the underlying business object.|
| `DataFormEditorLengthValidationRule` | Validates that the string length of the current value is between the specified minimum and maximum length. |
| `DataFormEditorRangeValidationRule` | Validates that the current value is between the specified minimum and maximum value. |
| `DataFormEditorRegexValidationRule` | Executes a custom regular expression to validate the current string value. |
| `DataFormEditorRequiredValidationRule` | Validates that the current value is not null or an empty string. |

> > When using this approach and in addition, the validation is added to the properties defined in the `ViewModel` using Data Annotation, the validation set in the `ViewModel` is with high priority (you won't be able to remove, modify the validation in the `EditorGenerated` event).

Reviewer: I keep reading this part and I find it very hard to understand. I highly recommend rewriting it for clarity. You might need to split it into several sentences or whatever works best.

| `DataFormEditorRegexValidationRule` | Executes a custom regular expression to validate the current string value. |
| `DataFormEditorRequiredValidationRule` | Validates that the current value is not null or an empty string. |

> > When using runtime validation approach and you have defined a validation to the `ViewModel` (by setting the data annotations)&mdash;the validation defined in the `ViewModel` will be applied instead of the runtime validation. The [Data Annotation Validation]({%slug dataform-data-annotations%}) is with higher priority.

Reviewer: ```suggestion
> When using both runtime validation and `ViewModel` validation (via data annotations)&mdash;the validation defined in the `ViewModel` will be applied instead of the runtime validation. The [Data Annotation Validation]({%slug dataform-data-annotations%}) has the higher priority.
```
